### PR TITLE
(8460) Update spoke_vpc resource to include gcp_support

### PR DESCRIPTION
**Pull request by iggynopolis**
Update spoke_vpc resource to include gcp_support

### CODE REVIEW

**Comment by iggynopolis** [comment]
id=8460

**Comment by arvindsree** [comment]
add mantis ID in PR title

File: aviatrix/resource_spoke_vpc.go [hunk 1]
```diff
@@ -186,22 +189,23 @@ func resourceAviatrixSpokeVpcCreate(d *schema.ResourceData, meta interface{}) er
...
+			tags := &goaviatrix.Tags{
+				CloudType:    1,
+				ResourceType: "gw",
+				ResourceName: d.Get("gw_name").(string),
+				TagList:      gateway.TagList,
+			}
+			err = client.AddTags(tags)
+			if err != nil {
+				return fmt.Errorf("failed to add tags: %s", err)
+			}
```
Comment (arvindsree) [inline aviatrix/resource_spoke_vpc.go]: show error in else case that tags are only supported for aws cloud type 1

**Comment by arvindsree** [comment]
We need to add acceptance tests for this

File: aviatrix/resource_spoke_vpc.go [hunk 2]
```diff
@@ -122,16 +122,18 @@ func resourceAviatrixSpokeVpcCreate(d *schema.ResourceData, meta interface{}) er
...
 	if gateway.EnableNAT != "yes" {
 		gateway.EnableNAT = "no"
 	}
-	if cloudType := d.Get("cloud_type").(int); cloudType == 1 {
+	if gateway.CloudType == 1 || gateway.CloudType == 4 {
 		gateway.VnetRsrcGrp = ""
 		d.Set("vnet_and_resource_group_names", gateway.VnetRsrcGrp)
-	}
-	if cloudType := d.Get("cloud_type").(int); cloudType == 8 {
+	} else if gateway.CloudType == 8 {
```
Comment (arvindsree) [inline aviatrix/resource_spoke_vpc.go]: Is this logic correct? Isnt azure cloud type 8?

**Pull request merged**